subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DaemonJavaCompiler.java

```diff
@@ -69,6 +71,16 @@ protected DaemonForkOptions toDaemonForkOptions(JavaCompileSpec spec) {
             .build();
     }
 
+    private File findSuitableExecutable(JavaCompileSpec spec) {
+        final ForkOptions forkOptions = spec.getCompileOptions().getForkOptions();
+        if (forkOptions.getExecutable() != null) {
+            return new File(forkOptions.getExecutable());
+        } else if (forkOptions.getJavaHome() != null) {
+            return new File(forkOptions.getJavaHome(), "bin/java");
```

Comment:
:o: You probably need to use `org.gradle.internal.os.OperatingSystem#getExecutableName` here.